### PR TITLE
Fix memory access issue with DDP

### LINKED ISSUE #166: issue while training using DDP
First of all, thank you for your great contributions to this community.

Unfortunately, I had a problem while training on a DDP with multiple GPUs :(
Training under a single GPU is well-performing, but when I tried training using 2 GPUs, an error occurred in the second GPU’s process. (I used NeighborhoodAttention2D and an error occurred in "attn = attn.softmax(dim=-1)")

An error occurred with the "NATTEN failure: cutlass error: Error Internal at: 102" messages.
As I searched, another user reported this error number and answered with a contiguous problem, but I think this one is not related to contiguous().

My running environment is organized as follows:
GPU: NVIDIA A6000 x2
Pytorch version: 2.4.1

### PR DESCRIPTION
**Pull request by alihassanijr**
Fixes #166.

According to the following PT forum post, extensions may have to use device guards in order to support DDP/multiprocess applications:

https://discuss.pytorch.org/t/cuda-illegal-memory-access-for-cuda-extension-in-multiprocessing/198767/2

Even though NATTEN hasn't had any issues with DDP since its creation, this issue can still break user applications, as it does in #166.

### CODE REVIEW

**Comment by alihassanijr** [comment]
TODOs:

- [ ] confirm #166 is fixed ( @minwoo-yu )
- [ ] doc fixes (unrelated to this, but in the process of debugging the issue, I noticed some inconsistencies in the docs regarding natten context; those need to be fixed asap.)
- [ ] bump version to 0.17.2
- [ ] build new binaries and push over pypi
